### PR TITLE
Integrating nodes for Path of Travel element from master to 2020

### PR DESCRIPTION
**Pull request by jrichter2019**
### Purpose

Integration to 2020 branch: Adding nodes for new 2020 functionality - Path of Travel.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers

@AndyDu1985

### CODE REVIEW

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      public void Create_ValidArgs()
+      {
+         var pathOfTravelOneToOne = PathOfTravel.ByFloorPlanPoints(
+            GetFloorPlan(),
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            new Point[] { Point.ByCoordinates(100, 100, 0) },
+            false);
+       
+         Assert.NotNull(pathOfTravelOneToOne);
+         Assert.Equals(pathOfTravelOneToOne.GetLength(0), 1);
```
Comment (ZiyunShang) [inline test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: Here Assert.Equals should not be used for Assertions. Should replace it with Assert.AreEqual like 
Assert.AreEqual(1, pathOfTravelOneToOne.GetLength(0));

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+         Assert.NotNull(pathOfTravelOneToOne[0]);
+
+         var pathOfTravelManyToMany = PathOfTravel.ByFloorPlanPoints(
+            GetFloorPlan(),
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            new Point[] { Point.ByCoordinates(100, 100, 0) },
+            true);
+
+         Assert.NotNull(pathOfTravelManyToMany);
+         Assert.Equals(pathOfTravelManyToMany.GetLength(0), 1);
```
Comment (ZiyunShang) [inline test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: The same as line 27.

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+               null,                                                 // null end point array
+               true));
+      }
+
+      [Test]
+      [TestModel(@".\Empty.rvt")]
+      public void Create_DifferentPointCounts()
+      {
+         Assert.Throws(
+            typeof(System.ArgumentException),
```
Comment (ZiyunShang) [inline test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: Is ArgumentException should be ArgumentNullException?

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      public void Create_PointsIdentical()
+      {
+         var pathOfTravelOneToOne = PathOfTravel.ByFloorPlanPoints(
+            GetFloorPlan(),
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            false);
+
+         Assert.NotNull(pathOfTravelOneToOne);
+         Assert.Equals(pathOfTravelOneToOne.GetLength(0), 1);
```
Comment (ZiyunShang) [inline test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: The same as line 27.

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+         Assert.Null(pathOfTravelOneToOne[0]);
+
+         var pathOfTravelManyToMany = PathOfTravel.ByFloorPlanPoints(
+            GetFloorPlan(),
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            true);
+
+         Assert.NotNull(pathOfTravelManyToMany);
+         Assert.Equals(pathOfTravelManyToMany.GetLength(0), 1);
```
Comment (ZiyunShang) [inline test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: The same as line 27.

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+      public void Create_PointsTooClose()
+      {
+         var pathOfTravelOneToOne = PathOfTravel.ByFloorPlanPoints(
+            GetFloorPlan(),
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            new Point[] { Point.ByCoordinates(double.Epsilon / 2, 0, 0) },
+            false);
+
+         Assert.NotNull(pathOfTravelOneToOne);
+         Assert.Equals(pathOfTravelOneToOne.GetLength(0), 1);
```
Comment (ZiyunShang) [inline test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: The same as line 27.

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+         Assert.Null(pathOfTravelOneToOne[0]);
+
+         var pathOfTravelManyToMany = PathOfTravel.ByFloorPlanPoints(
+            GetFloorPlan(),
+            new Point[] { Point.ByCoordinates(0, 0, 0) },
+            new Point[] { Point.ByCoordinates(double.Epsilon / 2, 0, 0) },
+            true);
+
+         Assert.NotNull(pathOfTravelManyToMany);
+         Assert.Equals(pathOfTravelManyToMany.GetLength(0), 1);
```
Comment (ZiyunShang) [inline test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: The same as line 27.

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs (hunk 3 above)
Comment (jrichter2019) [reply test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: No, the arguments are valid objects, just a different number of points in each array.

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs (hunk 1 above)
Comment (jrichter2019) [reply test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: Thanks, fixed in a new commit.

**Comment by ZiyunShang** [comment]
@jrichter2019 I think you could run your node tests with RTF to make sure they are working.
You can get RTF from https://github.com/DynamoDS/RevitTestFramework .

File: test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs (hunk 3 above)
Comment (ZiyunShang) [reply test/Libraries/RevitNodesTests/Elements/PathOfTravelTests.cs]: But your startPtList has no element, a ArgumentNullException would throw first here https://github.com/DynamoDS/DynamoRevit/blob/534b165555ab473ab7a4a1f08a900c33ebce8820/src/Libraries/RevitNodes/Elements/PathOfTravel.cs#L77 .
I think it couldn't get a ArgumentExcetion in this test.

**Comment by ZiyunShang** [comment]
We have a new branch RC2.2.1_Revit2020 for Revit 2020.1 release, this change need not to cherry pick to RC2.2.0_Revit2020 branch now.

**Pull request closed without merging**